contracts/foundation/service_provider.go

```diff
@@ -1,8 +1,15 @@
 package foundation
 
 type ServiceProvider interface {
-	//Register any application services.
+	// Register any application services.
 	Register(app Application)
-	//Boot any application services after register.
+	// Boot any application services after register.
 	Boot(app Application)
 }
+
+// UnimplementedServiceProvider is a default implementation of the Provider interface.
+type UnimplementedServiceProvider struct{}
```

Comment:
Q: Is it better to change `UnimplementedServiceProvider` to `BaseServiceProvider`?

Reply:
Optimized